File: src/main/java/com/sica/backend/repository/EmployeeRepository.java
package com.sica.backend.repository;

import com.sica.backend.entity.Employee;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface EmployeeRepository extends JpaRepository<Employee, String> {
    Optional<Employee> findByEmployeeNumber(String employeeNumber);
    Boolean existsByEmployeeNumber(String employeeNumber);
}

File: src/main/kotlin/com/sica/backend/repository/EmployeeRepository.kt
package com.sica.backend.repository

import com.sica.backend.entity.Employee
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface EmployeeRepository : JpaRepository<Employee, String> {
    fun findByEmployeeNumber(employeeNumber: String): Optional<Employee>
    fun existsByEmployeeNumber(employeeNumber: String): Boolean
}
